crypto/ec/build.info

```diff
@@ -71,6 +71,14 @@ SOURCE[../../providers/libfips.a]=$COMMON
 # need to be applied to all affected libraries and modules.
 DEFINE[../../libcrypto]=$ECDEF
 DEFINE[../../providers/libfips.a]=$ECDEF
+DEFINE[../../providers/libdefault.a]=$ECDEF
+# We only need to include the ECDEF stuff in the legacy provider when
+# it's a separate module and it's dynamically linked with libcrypto.
+# Otherwise, it already gets everything that the static libcrypto.a
+# has, and doesn't need it added again.
+IF[{- !$disabled{module} && !$disabled{shared} -}]
+  DEFINE[../providers/liblegacy.a]=$ECDEF
```

Comment:
Yes, I know that we currently do not need these flags for the legacy provider.  However, they do not hurt and if we ever move a curve to the legacy provider we at least move the whole implementation (including the ASM part).  That is one of the reasons for this change.  The other is for symmetry with the remaining build.info files.